cpp/src/arrow/adapters/orc/adapter_options.h

```diff
@@ -0,0 +1,228 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <set>
+#include <sstream>
+
+#include "arrow/adapters/orc/adapter_common.h"
+#include "arrow/io/interfaces.h"
+#include "arrow/util/visibility.h"
+#include "orc/OrcFile.hh"
+
+namespace liborc = orc;
+
+namespace arrow {
+
+namespace adapters {
+
+namespace orc {
+// Components of ORC Writer Options
+
+/**
+ * Options for creating a Writer.
+ */
+class ARROW_EXPORT WriteOptions {
+ public:
+  WriteOptions();
+  WriteOptions(const WriteOptions&);
+  WriteOptions(WriteOptions&);
+  WriteOptions& operator=(const WriteOptions&);
+  virtual ~WriteOptions();
+
+  /**
+   * Get the ORC writer options
+   * @return The ORC writer options this WriteOption encapsulates
+   */
+  std::shared_ptr<liborc::WriterOptions> GetOrcWriterOptions() const {
+    return orc_writer_options_;
+  }
+
+  /**
+   * Set the batch size.
+   */
+  WriteOptions& SetBatchSize(uint64_t size);
+
+  /**
+   * Get the batch size.
+   * @return if not set, return default value.
+   */
+  uint64_t GetBatchSize() const;
+
+  /**
+   * Set the stripe size.
+   */
+  WriteOptions& SetStripeSize(uint64_t size);
+
+  /**
+   * Get the stripe size.
+   * @return if not set, return default value.
+   */
+  uint64_t GetStripeSize() const;
```

Comment:
Unless really useful and/or necessary, we tend to use signed integers, e.g. `int64_t` here.

Reply:
Sure. In the converter I just used a bunch of `static_cast` to cast Arrow `int64_t`s into ORC `uint64_t`s.